pkg/controller/installation/installation_controller.go

```diff
@@ -853,6 +865,37 @@ func (r *ReconcileInstallation) addCustomInformer(crd runtime.Object, namespace
 	return nil
 }
 
+func checkEnvVars(checks map[string]func(string, bool) error) error {
+	for env, check := range checks {
+		value, exists := os.LookupEnv(env)
+		if err := check(value, exists); err != nil {
+			return fmt.Errorf("validation failure for env var %s: %w", env, err)
+		}
+	}
+
+	return nil
+}
+
+func optionalEnvVar(check func(string) error) func(string, bool) error {
+	return func(value string, ok bool) error {
+		if !ok {
+			return nil
+		}
+
+		return check(value)
+	}
+}
+
+func requiredEnvVar(check func(string) error) func(string, bool) error {
```

Comment:
Hey @sergioifg94 I am little bit confused by the syntax of the checkEnvVars call at line 599, but can't see the requiredEnvVar function being used anywhere? Could you explain what am I missing please?

Reply:
`checkEnvVars` is a little function to validate the env vars during the preflight checks, so the installation doesn't progress if an invalid variable is passed. It accepts a map in the form _Env var_ -> _Validating function_, so for now I'm passing the `FORCE_ZONE_DISTRIBUTION`, and a function to check that the value can be parsed as `bool`.

The `optionalEnvVar` and `requiredEnvVar` functions are utility closures for the sake of re utilization. I left `requiredEnvVar` there even though it's unused in case it's needed at some point

Reply:
Thanks Sergio

Reply:
@MStokluska @laurafitzgerald fyi this all could be implemented in a more OOP fashion that might be easier to understand (but more boilerplate code imo):

```go
type EnvVarValidation interface {
   CheckEnvVar(value string, ok bool) error
}

func checkEnvVars(validations map[string]EnvVarValidation) error {
   for env, check := range checks {
    value, exists := os.LookupEnv(env)
    if err := check.CheckEnvVar(value, exists); err != nil {
      return fmt.Errorf("validation failure for env var %s: %w", env, err)
    }
  }

  return nil
}
```

Instead of a function an interface, which would require to implement with structs

```go

type OptionalEnvVar struct {
  InnerCheck EnvVarValidation
}

func (c *OptionalEnvVar) CheckEnvVar(value string, ok bool) error {
  if !ok {
    return nil
  }

  return c.InnerCheck.CheckEnvVar(value, ok)
}

type BoolEnvVar struct {}

func (c *BoolEnvVar) CheckEnvVar(value string, ok bool) error {
  _, err := strconv.ParseBool(value)
  return err
}
```

So the map would look like

```go
map[string]EnvVarValidation {
  resources.AntiAffinityRequiredEnvVar: &OptionalEnvVar{
    InnerCheck: &BoolEnvVar{},
  } 
}
```